src/transformers/pipelines/text_to_speech.py

```diff
@@ -0,0 +1,168 @@
+
+    See the list of available models on [huggingface.co/models](https://huggingface.co/models?filter=text-to-speech).
+    """
+
+    def __init__(self, *args, vocoder=None, processor=None, sampling_rate=None, sample_rate_name=None, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        if self.framework == "tf":
+            raise ValueError("The TextToSpeechPipeline is only available in PyTorch.")
+
+        self.model_type = self.model.config.model_type
+
+        # legacy boolean to check particular model type speecht5
+        # for which there are no SpeechT5ForTextToSpeechWithHifiGanHead
+        self.is_speecht5 = self.model_type == "speecht5"
+
+        # check if the model only takes one speaker embeddings per generation
+        if self.model_type in ONLY_ONE_SPEAKER_EMBEDDINGS_LIST:
+            self.only_one_speaker_embeddings = True
+        else:
+            self.only_one_speaker_embeddings = False
+
+        if self.is_speecht5 and vocoder is None:
+            raise ValueError(
+                """vocoder is None but speecht5 is used.
+                              Try passing a repo_id or an instance of SpeechT5HifiGan."""
+            )
+        elif self.is_speecht5 and isinstance(vocoder, str):
+            vocoder = SpeechT5HifiGan.from_pretrained(vocoder).to(self.model.device)
+        elif self.is_speecht5 and not isinstance(vocoder, SpeechT5HifiGan):
+            raise ValueError(
+                """Must pass a valid vocoder to the TTSPipeline if speecht5 is used.
+                              Try passing a repo_id or an instance of SpeechT5HifiGan."""
+            )
+
+        self.processor = processor
+        self.vocoder = vocoder
+
+        if self.is_speecht5:
+            self.sampling_rate = vocoder.config.sampling_rate
+        elif sampling_rate is None:
+            # get sampling_rate from config and generation config
+            self.sampling_rate = None
+
+            config = self.model.config.to_dict()
+            gen_config = self.model.__dict__.get("generation_config", None)
+            if gen_config is not None:
+                config.update(gen_config.to_dict())
+
+            for sampling_rate_name in ["sample_rate", "sampling_rate"]:
+                sampling_rate = config.get(sampling_rate_name, None)
+                if sampling_rate is not None:
+                    self.sampling_rate = sampling_rate
+        else:
+            self.sampling_rate = sampling_rate
+
+    def preprocess(self, text, speaker_embeddings=None, **kwargs):
+        if self.is_speecht5:
+            inputs = self.processor(text=text, return_tensors="pt")
+            if speaker_embeddings is None:
+                embeddings_dataset = load_dataset("Matthijs/cmu-arctic-xvectors", split="validation")
+                speaker_embeddings = torch.tensor(embeddings_dataset[7305]["xvector"]).unsqueeze(0)
+
+            return {"input_ids": inputs["input_ids"], "speaker_embeddings": speaker_embeddings}
```

Comment:
For now, only `speaker_embeddings` is taken into account, but future models could have other conditional inputs.